app/components/hook-switch.js

```diff
@@ -13,12 +18,12 @@ export default Ember.Component.extend({
       let pusher = this.get('pusher'),
         repoId = hook.get('id');
 
-      return hook.toggle().then(() => {
+      yield hook.toggle().then(() => {
         pusher.subscribe(`repo-${repoId}`);
       }, () => {
         this.toggleProperty('hook.active');
         return this.sendAction('onToggleError', hook);
       });
     }
-  }
+  }),
```

Comment:
I don’t think there would be any effect between hook switches, as each component is isolated. I was more thinking about what would happen if someone clicked a switch multiple times. But when I try clicking a switch when it’s already saving, it does nothing, so never mind! 😀